### PR TITLE
added basic structure

### PR DESCRIPTION
**Pull request by makovii**

### CODE REVIEW

File: identity-backend/src/user/repository/customer.model.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
+import mongoose, { HydratedDocument } from 'mongoose';
+import { Auth } from '../../auth/repository/auth.model';
+
+export type UserDocument = HydratedDocument<User>;
+
```
Comment (Er4ik) [inline identity-backend/src/user/repository/customer.model.ts]: Why you named the file as a cutomer, but Model named as User?

File: identity-backend/src/user/repository/customer.repository.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    
+    let exist = await this.userModel.findOne({ auth_id: id });
+    if (!exist) {
+      newUser = await this.userModel.create({ auth_id: id });
+    } else {
+      newUser = exist;
+    }
+
+    return newUser;
+  }
```
Comment (Er4ik) [inline identity-backend/src/user/repository/customer.repository.ts]: Written a bit in complicated way. You use mutable variables, that is not as good as not mutable.
I guess could be better to use something like this:
`const user = await this.userModel.findOne({ auth_id: id });`
`if (!user) {`
`    return await this.userModel.create({ auth_id: id });`
`}`
`return user;`

File: identity-backend/src/user/controller/dto/user.dto.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+
+export class UserDto {
+  // auth_id: Auth;
```
Comment (Er4ik) [inline identity-backend/src/user/controller/dto/user.dto.ts]: Why this line is commented?

Comment (makovii) [reply identity-backend/src/user/controller/dto/user.dto.ts]: forgot to remove

File: identity-backend/src/user/repository/customer.repository.ts (hunk 2 above)
Comment (makovii) [reply identity-backend/src/user/repository/customer.repository.ts]: yes, at this stage your solution will be better. But let's say we don't want to show the user all the fields that we store in the database. Mappers are used for this.
I have updated the file

File: identity-backend/src/user/repository/customer.model.ts (hunk 1 above)
Comment (makovii) [reply identity-backend/src/user/repository/customer.model.ts]: my bad. Already fixed

**Review by Er4ik — APPROVED** [review]

**Pull request merged**